src/jit/lsraxarch.cpp

```diff
@@ -281,12 +281,12 @@ void Lowering::TreeNodeInfoInit(GenTree* tree)
 
                 if (cmpOp1->IsSIMDEqualityOrInequality() && (cmpOp2->IsIntegralConst(0) || cmpOp2->IsIntegralConst(1)))
                 {
-                    // We always generate code for a SIMD equality comparison, but the compare
-                    // is contained (evaluated as part of the GT_JTRUE).
+                    // We always generate code for a SIMD equality comparison, but the compare itself produces no value.
                     // Neither the SIMD node nor the immediate need to be evaluated into a register.
                     l->clearOperandCounts(cmp);
                     l->clearDstCount(cmpOp1);
                     l->clearOperandCounts(cmpOp2);
+                    cmpOp2->SetAsContained();
```

Comment:
Hmm, it seems that even after this change we still can't move this logic out of `TreeNodeInfoInit` easily.